editions/tw5.com/tiddlers/system/version-macros.tid

```diff
@@ -1,12 +1,19 @@
 code-body: yes
 created: 20161008085627406
-modified: 20221007122259593
+modified: 20231205095748960
 tags: $:/tags/Macro
 title: $:/editions/tw5.com/version-macros
 type: text/vnd.tiddlywiki
 
-\define .from-version(version)
-<$link to={{{ [<__version__>addprefix[Release ]] }}} class="doc-from-version">{{$:/core/images/warning}} New in: <$text text=<<__version__>>/></$link>
+\procedure .from-version-template(class, text)
+<$link to={{{ [<version>addprefix[Release ]] }}} class=<<class>> >{{$:/core/images/info-button}} <<text>><$text text=<<version>>/></$link>
+\end
+
+\procedure .from-version(version)
+<$let lastNewVersion="5.3.0">
```

Comment:
IMO there should be **no** manual modification needed, when versions change. So I think that does not really work out. TW has a `<<version>>` macro, which can be used to compare between versions.

Reply:
What programmatic criteria about what is "new" do you suggest?

* Only the latest - less informative in situations such as now, where we have a bunch of new things from v5.3.0, but the bugfix v5.3.1 is the latest.
* Latest minor release - informative now, since v5.3.0 introduces a lot of new things, but might get less informative over time if we stay on v5.3.x for a long time.
* A fixed number of releases?
* A fixed time between releases? E.g. everything released in the last 6 months or so. "Release" tiddlers have a `released` field, so this should be easily doable.

Reply:
A simpler version could be to use a global variable like.

```
\procedure .from-version-reference() 5.3.0
```

So it would be more obvious what it is used for. It still would be a manual change an extra maintenence burden. I'm not sure how often it would be needed to change that global variable